lib/widgets/new_expense.dart

```diff
@@ -0,0 +1,305 @@
+  Widget _buildTitleTextField() {
+    return TextField(
+      maxLength: 50,
+      keyboardType: TextInputType.text,
+      controller: _titleController,
+      decoration: const InputDecoration(labelText: 'Title'),
+    );
+  }
+
+  /// Builds a [TextField] widget for entering the amount of a new expense.
+  /// The [TextField] has a [TextInputType.number] keyboard type and a [_amountController] controller.
+  /// It also has a label text of 'Amount' and a prefix text of '\$' in the decoration.
+  Widget _buildAmountTextField() {
+    return TextField(
+      keyboardType: TextInputType.number,
+      controller: _amountController,
+      decoration: const InputDecoration(labelText: 'Amount', prefixText: '\$'),
+    );
+  }
+
+  /// Builds a dropdown button widget for selecting a category.
+  ///
+  /// Returns a [DropdownButton] widget that displays a list of [Category] values
+  /// as dropdown menu items. The currently selected category is stored in the
+  /// [selectedCategory] variable. When the user selects a new category, the
+  /// [onChanged] callback function is called with the new value, which updates
+  /// the selected category and rebuilds the widget.
+  Widget _buildCategorySelectionDropdown() {
+    return DropdownButton(
+      value: selectedCategory,
+      items: Category.values
+          .map((item) => DropdownMenuItem(
+                value: item,
+                child: Text(item.name.toUpperCase()),
+              ))
+          .toList(),
+      onChanged: (value) {
+        if (value != null) {
+          setState(() {
+            selectedCategory = value;
+          });
+        }
+      },
+    );
+  }
+
+  /// Builds a row containing a text widget displaying the selected date formatted using [formatter]
+  /// and an icon button that opens a date picker dialog when pressed.
+  /// Returns the row as a [Widget].
+  Widget _buildDatePickerRow() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Text(formatter.format(selectedDate)),
+        IconButton(
+          onPressed: _presentDatePicker,
+          icon: const Icon(
+            Icons.calendar_month,
+          ),
+        ),
+      ],
+    );
+  }
+
+  /// Builds a cancel button widget.
+  ///
+  /// Returns a [TextButton] widget with the text "Cancel" and an [onPressed] function that pops the current route off the navigator.
+  Widget _buildCancelButton() {
+    return TextButton(
+      onPressed: () {
+        Navigator.pop(context);
+      },
+      child: const Text('Cancel'),
+    );
+  }
+
+  /// Builds a save expense button widget.
+  ///
+  /// Returns an [ElevatedButton] widget with the text 'Save Expense' and
+  /// an onPressed callback that calls [_submitExpenseData] function.
+  Widget _buildSaveExpenseButton() {
+    return ElevatedButton(
+      onPressed: _submitExpenseData,
+      child: const Text('Save Expense'),
+    );
```

Comment:
Once again great comments here and the update expense function was a good addition to the app